src/ixdat/readers/zilien.py

```diff
@@ -162,150 +177,279 @@ def read(self, path_to_file, cls=ECMSMeasurement, name=None, **kwargs):
+            for column_header in column_headers
+        ]
+
+        # Split rows according to techniques used according to experiments
+        # Combine the experiment numbers and the technique numbers
+        # in order to create unique identifiers for a successful split
+        exp_nums = data[:count, column_headers.index("experiment_number")]
+        tech_nums = data[:count, column_headers.index("technique_number")]
+        split_vector = exp_nums * 1000 + tech_nums
+        splits = self._get_biologic_splits(split_vector)
+
+        # Create Ixdat series
+        column_series = []
+        for begin, end in splits:
+            column_series += self._create_series_objects(
+                column_headers, names_and_units, data[begin:end, :]
+            )
+
+        return column_series
+
+    def _create_series_objects(self, column_headers, names_and_units, data):
+        """Create an Ixdat series objects from a given portion of a dataset."""
+
+        series_objects = []
+        time_series = None
+
+        for column_number, column_header in enumerate(column_headers):
+            # Skip meta columns in the EC-lab dataset
+            if column_header in ("experiment_number", "technique_number"):
+                continue
+
+            column_data = data[:, column_number]
+
+            # Skip holes in the EC-lab dataset
+            if all(np.isnan(column_data)):
+                continue
+
+            # Form series kwargs
+            series_name, unit, standard_name = names_and_units[column_number]
             series_kwargs = {
                 "name": series_name,
                 "unit_name": unit,
                 "data": column_data,
             }
-            if column_header == "Time [s]":  # Form TimeSeries
-                column_series = TimeSeries(**series_kwargs, tstamp=timestamp)
-                last_time_series = column_series
+
+            # Create the series
+            if column_header in ("Time [s]", "time/s"):
+                series_object = TimeSeries(**series_kwargs, tstamp=self._timestamp)
+                time_series = series_object
             else:
-                column_series = ValueSeries(**series_kwargs, tseries=last_time_series)
+                assert (
+                    time_series is not None
+                ), "Time column must be first in a dataset series."
+                series_object = ValueSeries(**series_kwargs, tseries=time_series)
 
-            series.append(column_series)
+            series_objects.append(series_object)
 
-        return series, aliases
+        return series_objects
+
+    # --- UTILS ---
+
+    @staticmethod
+    def _get_series_splits(series_headers):
+        """Series names and their index pairs in the whole read dataset.
+
+        Args:
+            series_headers (list): Series name headers (even empty).
+
+        Returns:
+            list, list: List of tuples with index pairs
```

Comment:
er list, list or list of tuples?

Reply:
Both are lists of tuples. Do you think I should write out all types the return value consists of?

Reply:
Ah, my bad, I read read it wrong. No, don't write the more out, this is fine.